sdcm/utils/common.py

```diff
@@ -355,13 +392,61 @@ def inner(*args, **kwargs):
                 return_val = fun(*args, **kwargs)
                 LOGGER.debug("[{thread_name}] Done.".format(thread_name=thread_name))
                 return return_val
+            inner.__name__ = fun.__name__
             return inner
 
+        results = []
         with ThreadPoolExecutor(max_workers=self.num_workers) as pool:
             LOGGER.debug("Executing in parallel: '{}' on {}".format(func.__name__, self.objects))
             if not self.disable_logging:
                 func = func_wrap(func)
-            return list(pool.map(func, self.objects, timeout=self.timeout))
+            futures = []
+
+            for obj in self.objects:
```

Comment:
why not do all this inside the wrapper function, inside this error prone loops ?

Reply:
@fruch could you explain what you mean, i don't sure i understand what you mean.
Current implementation allow to run func in different thread, and process exception independently  for each future. Especially for TimeoutException which could happened for future.

Reply:
I thought it would be clearer to handle everything inside the wrapper, and only handle TimeoutException outside of it.
but this isn't that importent